### ISSUE TITLE
This isn't generic enough

### ISSUE THREAD

**Issue by toastal** [edkv/elm-generic-dict#1]
For the most part, everything could be `%s/String/comparable/g`. Number types are valid keys in Elm's Dict so you've restricted this library more than it needs to be.

**Comment by Erudition** [comment]
I think the main non-comparable thing I want to use is Custom types, but I can totally make them comparable with the conversion function so it's unfortunate that this library doesn't allow us to take advantage of order.

**Comment by toastal** [comment]
Yes, Elm is limited. But it's still worthwhile to abstract this as much as possible.

**Comment by Erudition** [comment]
I was talking about the limitations of this library, not Elm. 
If I simply want to use `WidgetID Int` as my keys, I could, thanks to this library, and with unboxing there would be no compiled difference from using plain `Int`s - so why not take advantage of additional type safety.

But this library e.g. removes `foldl` and `foldr` because they don't want to assume your `comparable` type is also `ordered`. Given the example above is, I think, the main use case for custom keys in dictionaries, it's unfortunate we can't have all the same functionality as a classic `Dict`.

**Comment by mihkell** [comment]
https://github.com/turboMaCk/any-dict/blob/2.6.0/src/Dict/Any.elm This looks to be generic enough.

**Comment by Erudition** [comment]
Of course AnyDict is great, but the point of this generic Dict is to not contain a function in the data structure, the issue with AnyDict.

Here is a breakdown of the problems:
https://package.elm-lang.org/packages/pzp1997/assoc-list/latest/
